src/graphics/program-lib/chunks/msdf.frag

```diff
@@ -33,19 +33,27 @@ uniform float font_sdfIntensity; // intensity is used to boost the value read fr
 uniform float font_pxrange;      // the number of pixels between inside and outside the font in SDF
 uniform float font_textureWidth; // the width of the texture atlas
 
-vec4 applyMsdf(vec4 color) {
-    float font_size = 16.0; // TODO fix this
+uniform vec4 outline_color;
+uniform float outline_distance;
+uniform vec4 shadow_color;
+uniform vec2 shadow_offset;
+//vec2 shadowVector = vec2(0.0062, -0.0046);
 
+vec4 applyMsdf(vec4 color) {
     // sample the field
     vec3 tsample = texture2D(texture_msdfMap, vUv0).rgb;
+    vec2 uvShdw = vUv0 - shadow_offset;
+    vec3 ssample = texture2D(texture_msdfMap, uvShdw).rgb;
     // get the signed distance value
     float sigDist = median(tsample.r, tsample.g, tsample.b);
+    float sigDistShdw = median(ssample.r, ssample.g, ssample.b);
 
     #ifdef USE_FWIDTH
         // smoothing depends on size of texture on screen
         vec2 w = fwidth(vUv0);
-        float smoothing = clamp(map(0.0, 2.0 * font_pxrange / font_textureWidth, w.x), 0.0, 0.5);
+        float smoothing = clamp(w.x * font_textureWidth / font_pxrange, 0.0, 0.5);
     #else
+        float font_size = 16.0; // TODO fix this
```

Comment:
Does this need to be fixed before we merge?

Reply:
Old code, moved to a fallback branch where it's only being used.